src/network/networkd-iprule.c

```diff
@@ -0,0 +1,403 @@
+
+        link_ref(link);
+
+        return 0;
+}
+
+int config_parse_ip_rule_tos(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_ip_rule_free_ IPRule *n = NULL;
+        Network *network = userdata;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = ip_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = safe_atou(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse ip rule tos, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        n->tos = k;
+        n = NULL;
+
+        return 0;
+}
+
+int config_parse_ip_rule_table(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_ip_rule_free_ IPRule *n = NULL;
+        Network *network = userdata;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = ip_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = safe_atou(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse ip rule tos, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        n->table = k;
+        n = NULL;
+
+        return 0;
+}
+
+int config_parse_ip_rule_fwmark_mask(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_ip_rule_free_ IPRule *n = NULL;
+        Network *network = userdata;
+        uint32_t k;
+        char *p;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = ip_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        p = strchr(rvalue, '/');
```

Comment:
Does this even compile without warnings? `rvalue` is `const`, and you're modifying it. That's wrong.